docs/data-sources/eg_event_sources.md

```diff
@@ -0,0 +1,97 @@
+---
+subcategory: "EventGrid (EG)"
+layout: "huaweicloud"
+page_title: "HuaweiCloud: huaweicloud_eg_event_sources"
+description: |-
+  Use this data source to filter EG event sources within HuaweiCloud.
+---
+
+# huaweicloud_eg_event_sources
+
+Use this data source to filter EG event sources within HuaweiCloud.
+
+## Example Usage
+
+### Query all kinds of event sources
+
+```hcl
+data "huaweicloud_eg_event_sources" "test" {}
+```
+
+### Query all official event sources
+
+```hcl
+data "huaweicloud_eg_event_sources" "test" {
+  provider_type = "OFFICIAL"
+}
+```
+
+### Query the custom event source with the specified name
+
+```hcl
+variable "source_name" {}
+
+data "huaweicloud_eg_event_sources" "test" {
+  provider_type = "CUSTOM"
+  name          = var.source_name
+}
+```
+
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region where the event sources are located.  
+  If omitted, the provider-level region will be used.
+
+* `provider_type` - (Optional, String) Specifies the type of the event sources to be queried.
```

Comment:
Please list the enumeration values ​​for this field.

Reply:
all enums are supported.